ddp-workspace/projects/ddp-dsm-ui/src/app/participant-list/participant-list.component.ts

```diff
@@ -1077,10 +1077,10 @@ export class ParticipantListComponent implements OnInit {
       this.setSavedFilterColumnsIfSelected();
       return;
     }
-    const filteredColumns = this.extractDefaultColumns(this.selectedColumns);
-    Object.assign(this.selectedColumns, filteredColumns);
+    const filteredColumns = this.getDefaultColumns();
+    this.selectedColumns = filteredColumns;
```

Comment:
The main cause of the bug was here. The `Objects.assign` just copied over everything from filteredColumns to selectedColumns, without removing what was in the selectedColumn before. So the old survey columns stayed there